### PR TITLE
[SPARK-46472][PYTHON][DOCS] Refine docstring of `array_prepend/array_append/array_insert`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr refine docstring of `array_prepend/array_append/array_insert` and add some new examples.

### Why are the changes needed?
To improve PySpark documentation

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass Github Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -379,6 +379,7 @@ jobs:
       SKIP_PACKAGING: true
       METASPACE_SIZE: 1g
       BRANCH: ${{ inputs.branch }}
+      SPARK_ANSI_SQL_MODE: true
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:382]: Check ansi is true , will revert this file after the test passes.

**Review by zhengruifeng — APPROVED** [review]
thanks for working on this

File: python/pyspark/sql/functions/builtin.py [hunk 2]
```diff
@@ -12993,14 +13051,66 @@ def array_insert(arr: "ColumnOrName", pos: Union["ColumnOrName", int], value: An
...
+    +------------------------+
+
+    Example 4: Inserting a NULL value
+
+    >>> from pyspark.sql import functions as sf
+    >>> from pyspark.sql.types import StringType
+    >>> df = spark.createDataFrame([(['a', 'b', 'c'],)], ['data'])
+    >>> df.select(sf.array_insert(df.data, 2, sf.lit(None).cast(StringType()))
+    ...   .alias("result")).show()
+    +---------------+
```
Comment (LuciferYang) [inline python/pyspark/sql/functions/builtin.py:13094]: image

It seems that the headers of the results from the connect and sql tests are inconsistent, try giving it an `alias`.

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (LuciferYang) [reply .github/workflows/build_and_test.yml:382]: image
Passed

**Review by HyukjinKwon — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged into master for Spark 4.0. Thanks @HyukjinKwon @zhengruifeng

**Pull request closed without merging**